appwidget-configuration/src/main/java/com/google/android/glance/appwidget/configuration/AppWidgetConfigurationScaffold.kt

```diff
@@ -131,6 +129,12 @@ class AppWidgetConfigurationState(
             internalState
         }
         instance.update(activity, glanceId)
+        
+        val intent = activity.intent.apply {
+            putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
```

Comment:
I don't see how we could reach this status, where the GlanceID was not null but we are missing the appWidgetID in the intent.

The GlanceID is extracted from the intent. There might be something else here. Are you modifying the intent?